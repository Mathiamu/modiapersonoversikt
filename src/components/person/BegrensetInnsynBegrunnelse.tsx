import * as React from 'react';
import { BegrensetInnsynTyper } from '../../models/person/person';

interface Props {
    begrunnelseType: BegrensetInnsynTyper;
}

function BegrensetInnsynBegrunnelse({begrunnelseType}: Props) {
    switch (begrunnelseType) {
        case BegrensetInnsynTyper.Kode6:
            return <>Bruker har kode 6 adresse</>;
        case BegrensetInnsynTyper.Kode7:
            return <>Bruker har kode 7 adresse</>;
        case BegrensetInnsynTyper.EgenAnsatt:
            return <>Bruker faller inn under begrensning for egen ansatt</>;
        default:
            return <>Ukjent sikkerhetsbegrensing</>;
    }
}

export default BegrensetInnsynBegrunnelse;