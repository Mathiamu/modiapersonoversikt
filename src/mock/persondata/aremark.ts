import {
    AdresseBeskyttelse,
    EgenAnsatt,
    ForelderBarnRelasjon,
    ForelderBarnRelasjonRolle,
    FullmaktsRolle,
    Kjonn,
    LocalDate,
    LocalDateTime,
    Person,
    PersonStatus,
    SikkerhetstiltakType,
    SivilstandType,
    Skifteform
} from '../../app/personside/visittkort-v2/PersondataDomain';

export const barnAremark: ForelderBarnRelasjon[] = [
    {
        ident: '12345678910',
        rolle: ForelderBarnRelasjonRolle.BARN,
        navn: [
            {
                fornavn: 'BARN1',
                mellomnavn: null,
                etternavn: 'BARNESEN'
            }
        ],
        fodselsdato: ['2008-03-15' as LocalDate],
        alder: 13,
        adressebeskyttelse: [
            {
                kode: AdresseBeskyttelse.UGRADERT,
                beskrivelse: 'UGRADERT'
            }
        ],
        bostedAdresse: [
            {
                linje1: 'Islandsgate 49',
                linje2: '7693 Hovedøya',
                linje3: 'Norge'
            }
        ],
        personstatus: [
            {
                kode: PersonStatus.BOSATT,
                beskrivelse: 'BOSATT'
            }
        ]
    }
];

export const aremark: Person = {
    fnr: '10108000398',
    navn: [
        {
            fornavn: 'AREMARK',
            mellomnavn: null,
            etternavn: 'TESTFAMILIEN'
        }
    ],
    kjonn: [
        {
            kode: Kjonn.M,
            beskrivelse: 'Mann'
        }
    ],
    fodselsdato: ['1980-10-10' as LocalDate],
    alder: 41,
    dodsdato: [],
    bostedAdresse: [
        {
            linje1: 'Islandsgate 49',
            linje2: '7693 Hovedøya',
            linje3: 'Norge'
        }
    ],
    kontaktAdresse: [
        {
            linje1: 'Aremarkveien 1',
            linje2: '0320 Aremark',
            linje3: null
        }
    ],
    navEnhet: {
        id: '0219',
        navn: 'NAV Bærum'
    },
    statsborgerskap: [
        {
            land: {
                kode: 'NOR',
                beskrivelse: 'NORGE'
            },
            gyldigFraOgMed: '1980-10-10' as LocalDate,
            gyldigTilOgMed: null,
            erHistorisk: false
        }
    ],
    adressebeskyttelse: [
        {
            kode: AdresseBeskyttelse.KODE6,
            beskrivelse: 'Ingen'
        }
    ],
    sikkerhetstiltak: [
        {
            type: SikkerhetstiltakType.TFUS,
            gyldigFraOgMed: '2005-02-13' as LocalDate,
            gyldigTilOgMed: '2030-02-15' as LocalDate
        }
    ],
    erEgenAnsatt: EgenAnsatt.JA,
    personstatus: [
        {
            kode: PersonStatus.BOSATT,
            beskrivelse: 'Bosatt i Norge'
        }
    ],
    sivilstand: [
        {
            type: {
                kode: SivilstandType.GIFT,
                beskrivelse: 'Gift'
            },
            gyldigFraOgMed: '2005-12-12' as LocalDate
        }
    ],
    foreldreansvar: [
        {
            ansvar: 'felles',
            ansvarlig: {
                fornavn: 'Test',
                etternavn: 'Testesen',
                mellomnavn: null
            },
            ansvarsubject: {
                fornavn: 'Barn',
                etternavn: 'Barnesen',
                mellomnavn: null
            }
        }
    ],
    deltBosted: [
        {
            startdatoForKontrakt: '2000-10-10' as LocalDate,
            sluttdatoForKontrakt: '2005-05-07' as LocalDate,
            adresse: {
                linje1: `Adresseveien 1`,
                linje2: '0000 Aremark',
                linje3: null
            }
        }
    ],
    dodsbo: [
        {
            adressat: {
                advokatSomAdressat: {
                    kontaktperson: {
                        fornavn: 'Advokat',
                        mellomnavn: null,
                        etternavn: 'Navnesen'
                    },
                    organisasjonsnavn: 'Advokatkontoret Aremark',
                    organisasjonsnummer: '01234567891011'
                },
                organisasjonSomAdressat: null,
                personSomAdressat: null
            },
            adresse: {
                linje1: 'Elgelia 20, 0000 Aremark',
                linje2: null,
                linje3: null
            },
            registrert: '2010-02-02' as LocalDate,
            skifteform: Skifteform.OFFENTLIG
        }
    ],
    fullmakt: [
        {
            motpartsPersonident: '123456789',
            motpartsPersonNavn: {
                fornavn: 'Navn',
                mellomnavn: null,
                etternavn: 'Navnesen'
            },
            motpartsRolle: FullmaktsRolle.FULLMEKTIG,
            omraade: ['*'],
            gyldigFraOgMed: '2015-01-01' as LocalDate,
            gyldigTilOgMed: '2017-12-12' as LocalDate
        }
    ],
    vergemal: [
        {
            ident: '21042900076',
            navn: {
                fornavn: 'Simen',
                mellomnavn: null,
                etternavn: 'Solli'
            },
            vergesakstype: 'voksen',
            omfang: 'personligeOgOekonomiskeInteresser',
            embete: 'Fylkesmannen i Troms og Finnmark',
            gyldighetstidspunkt: '2016-03-27' as LocalDate,
            opphoerstidspunkt: '2017-06-30' as LocalDate
        }
    ],
    tilrettelagtKommunikasjon: {
        talesprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ],
        tegnsprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ]
    },
    telefonnummer: [
        {
            retningsnummer: {
                kode: '+47',
                beskrivelse: 'retningsnummer er norsk'
            },
            identifikator: '99009900',
            sistEndret: '2017-10-10T10:15:30' as LocalDateTime,
            sistEndretAv: 'TESTFAMILIEN AREMARK',
            prioritet: 1
        }
    ],
    kontaktOgReservasjon: {
        personident: '10108000398',
        reservasjon: 'true',
        epostadresse: {
            value: 'epost@nav.no',
            sistOppdatert: null,
            sistVerifisert: '2013-01-01' as LocalDate
        },
        mobiltelefonnummer: {
            value: '90000000',
            sistOppdatert: null,
            sistVerifisert: null
        }
    },
    bankkonto: {
        kontonummer: '1234567890',
        banknavn: 'DNB ASA',
        sistEndret: '2006-03-15T00:00:00' as LocalDateTime,
        sistEndretAv: '1010800 BD03',

        bankkode: null,
        swift: 'DNBANOKKXXX',
        landkode: null,
        adresse: {
            linje1: 'Bankveien 1,',
            linje2: '0357 Bankestad',
            linje3: null
        },
        valuta: {
            kode: 'NOK',
            beskrivelse: 'Norske kroner'
        }
    },
    forelderBarnRelasjon: barnAremark
};
