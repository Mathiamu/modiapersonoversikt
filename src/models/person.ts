import { Kodeverk } from './kodeverk';

export interface Person {
    navn: Navn;
    kjønn: Kjønn;
    geografiskTilknytning?: string;
    fødselsnummer: string;
    alder: number;
    diskresjonskode?: string;
    bankkonto?: Bankkonto;
    statsborgerskap?: string;
    personstatus: Bostatus;
    sivilstand: Kodeverk;
    familierelasjoner: Familierelasjon[];
}

export interface Navn {
    sammensatt: string;
    fornavn: string;
    mellomnavn: string;
    etternavn: string;
}

export interface Bankkonto {
    erNorskKonto: boolean;
    kontonummer: number;
    bank: string;
    sistEndret: string;
    sistEndretAv: string;
}

export interface Familierelasjon {
    harSammeBosted: boolean;
    rolle: Relasjonstype;
    tilPerson: {
        navn: Navn;
        alder: number;
        fødselsnummer: string;
        personstatus: Bostatus;
        kjonn: Kjønn
    };
}

export interface Bostatus {
    dødsdato?: string;
    bostatus?: string;
}

export enum BostatusTyper {
    Død = 'DØD',
    Utvandret = 'UTVA'
}

export enum Relasjonstype {
    Barn = 'BARN',
    Samboer = 'SAMBOER',
    Partner = 'PARTNER',
    Ektefelle = 'EKTE',
    Gift = 'GIFT'
}

export enum Sivilstand {
    Gift = 'GIFT',
    Ugift = 'UGIFT'
}

export enum Kjønn {
    Mann = 'M',
    Kvinne = 'K'
}

export function erDød(personstatus: Bostatus) {
    return personstatus.dødsdato || personstatus.bostatus === BostatusTyper.Død;
}

export function getBarnUnder21(familierelasjoner: Familierelasjon[]) {
    return familierelasjoner
        .filter(relasjon => relasjon.rolle === Relasjonstype.Barn)
        .filter(relasjon => relasjon.tilPerson.alder <= 21);
}

export function getPartner(person: Person) {
    const aktuelleRelasjoner = [
        Relasjonstype.Ektefelle,
        Relasjonstype.Gift,
        Relasjonstype.Partner,
        Relasjonstype.Samboer
    ];
    return person.familierelasjoner.find(relasjon => aktuelleRelasjoner.includes(relasjon.rolle));
}
