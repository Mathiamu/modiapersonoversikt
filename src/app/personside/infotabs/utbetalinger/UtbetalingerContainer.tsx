import * as React from 'react';
import Utbetalinger from './Utbetalinger';
import { AppState } from '../../../../redux/reducers';
import { connect, Dispatch } from 'react-redux';
import { RestReducer } from '../../../../redux/restReducers/restReducer';
import { UtbetalingerResponse } from '../../../../models/utbetalinger';
import Innholdslaster from '../../../../components/Innholdslaster';
import { STATUS } from '../../../../redux/restReducers/utils';
import { Action } from 'redux';
import { hentUtbetalinger, reloadUtbetalinger } from '../../../../redux/restReducers/utbetalinger';
import { default as Filtrering, FilterState, PeriodeValg } from './filter/Filter';
import { getFraDateFromFilter, getTilDateFromFilter } from './utils/utbetalingerUtils';
import theme from '../../../../styles/personOversiktTheme';
import styled from 'styled-components';
import { Innholdstittel, Undertekst, Undertittel } from 'nav-frontend-typografi';
import ErrorBoundary from '../../../../components/ErrorBoundary';
import moment = require('moment');

interface State {
    filter: FilterState;
}

const initialState: State = {
    filter: {
        periode: {
            radioValg: PeriodeValg.SISTE_30_DAGER,
            egendefinertPeriode: {
                fra: moment().subtract(1, 'year').toDate(),
                til: new Date()
            }
        },
        utbetaltTil: [],
        ytelser: []
    }
};

interface StateProps {
    utbetalingerReducer: RestReducer<UtbetalingerResponse>;
}

interface DispatchProps {
    hentUtbetalinger: (fødselsnummer: string, fra: Date, til: Date) => void;
    reloadUtbetalinger: (fødselsnummer: string, fra: Date, til: Date) => void;
}

interface OwnProps {
    fødselsnummer: string;
}

type Props = StateProps & DispatchProps & OwnProps;

export const utbetalingerMediaTreshold = '80rem';

const UtbetalingerArticle = styled.article`
  display: flex;
  align-items: flex-start;
  @media(max-width: ${utbetalingerMediaTreshold}) {
    display: block;
  }
  .visually-hidden {
    ${theme.visuallyHidden}
  }
`;

const FiltreringSection = styled.section`
  min-width: 18rem;
  flex-basis: 18rem;
`;

const UtbetalingerSection = styled.section`
  position: relative;
  flex-grow: 1;
  min-width: 35rem; // Tabellene begynner å wrappe ved bredder mindre enn dette
  @media not all and (max-width: ${utbetalingerMediaTreshold}) {
      margin-left: ${theme.margin.layout};
  }
`;

const ArenaLenkeStyle = styled.div`
  text-align: right;
  position: absolute;
  top: ${theme.margin.px20}
  right: ${theme.margin.px20}
`;

function ArenaLenke() {
    return (
        <ArenaLenkeStyle>
            <Undertekst>
                <a className="lenke">Meldinger/utbetalinger i Arena</a>
            </Undertekst>
        </ArenaLenkeStyle>
    );
}

class UtbetalingerContainer extends React.Component<Props, State> {

    constructor(props: Props) {
        super(props);
        this.state = {...initialState};
        this.onFilterChange = this.onFilterChange.bind(this);
        this.reloadUtbetalinger = this.reloadUtbetalinger.bind(this);
    }

    onFilterChange(change: Partial<FilterState>) {
        this.setState(
            (prevState) => { // Sender inn funksjon her for å motvirke race-conditions fra UtbetaltTilValg og YtelseValg
                return {
                    filter: {
                        ...prevState.filter,
                        ...change
                    }
                };
            }
        );
    }

    reloadUtbetalinger() {
        const fra = getFraDateFromFilter(this.state.filter);
        const til = getTilDateFromFilter(this.state.filter);
        this.props.reloadUtbetalinger(this.props.fødselsnummer, fra, til);
    }

    componentDidMount() {
        if (this.props.utbetalingerReducer.status === STATUS.NOT_STARTED) {
            this.props.hentUtbetalinger(
                this.props.fødselsnummer,
                getFraDateFromFilter(this.state.filter),
                getTilDateFromFilter(this.state.filter)
            );
        }
    }

    render() {
        return (
            <ErrorBoundary>
                <UtbetalingerArticle>
                    <Innholdstittel className="visually-hidden">Brukerens utbetalinger</Innholdstittel>
                    <FiltreringSection>
                        <Filtrering
                            filterState={this.state.filter}
                            onChange={this.onFilterChange}
                            hentUtbetalinger={this.reloadUtbetalinger}
                            utbetalingReducer={this.props.utbetalingerReducer}
                        />
                    </FiltreringSection>
                    <UtbetalingerSection>
                        <Undertittel className="visually-hidden">Filtrerte utbetalinger</Undertittel>
                        <Innholdslaster avhengigheter={[this.props.utbetalingerReducer]}>
                            <Utbetalinger
                                utbetalinger={this.props.utbetalingerReducer.data.utbetalinger}
                                filter={this.state.filter}
                            />
                        </Innholdslaster>
                        <ArenaLenke/>
                    </UtbetalingerSection>
                </UtbetalingerArticle>
            </ErrorBoundary>
        );
    }
}

function mapStateToProps(state: AppState): StateProps {
    return ({
        utbetalingerReducer: state.restEndepunkter.utbetalingerReducer
    });
}

function mapDispatchToProps(dispatch: Dispatch<Action>): DispatchProps {
    return {
        hentUtbetalinger: (fødselsnummer: string, fra: Date, til: Date) =>
            dispatch(hentUtbetalinger(fødselsnummer, fra, til)),
        reloadUtbetalinger: (fødselsnummer: string, fra: Date, til: Date) =>
            dispatch(reloadUtbetalinger(fødselsnummer, fra, til))
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(UtbetalingerContainer);
