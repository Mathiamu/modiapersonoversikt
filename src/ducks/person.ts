import { Person } from '../models/person';
import { Action, Dispatch } from 'redux';
import { FetchError, FetchSuccess, RestActions, STATUS } from './utils';

export interface PersonState {
    data: Person;
    status: STATUS;
    error: string | null;
}

const mockPerson = {
    fornavn: 'Aremark',
    etternavn: 'Testfamilien',
    fodselsnummer: '10108000398'
};

const reducerName = 'person/';

const actionNames = {
    PENDING: reducerName + STATUS.PENDING,
    OK: reducerName + STATUS.OK,
    ERROR: reducerName + STATUS.ERROR
};

const initialState: PersonState = {
    data: mockPerson,
    status: STATUS.NOT_STARTED,
    error: null
};

function mockFetch(fodselsnummer: string) {
    return new Promise((resolve, reject) => {
        setTimeout(() => { resolve(mockPerson); }, 2500);
    });
}

export function hentPerson(fodselsnummer: string) {
    return (dispatch: Dispatch<Action>) => {
        dispatch({ type: actionNames.PENDING});
        return mockFetch(fodselsnummer)
            .then((person) => {
                dispatch({type: actionNames.OK, data: person});
                return person;
            })
            .catch((err) => {
                dispatch({type: actionNames.ERROR, error: err});
            });
    };
}

export default function reducer(state: PersonState = initialState, action: RestActions<Person>): PersonState {
    switch (action.type) {
        case actionNames.PENDING:
            return {...state, status: STATUS.PENDING };
        case actionNames.OK:
            return {...state, status: STATUS.OK, data: (<FetchSuccess<Person>> action).data};
        case actionNames.ERROR:
            return {...state, status: STATUS.ERROR, error: (<FetchError> action).error};
        default:
            return state;
    }
}