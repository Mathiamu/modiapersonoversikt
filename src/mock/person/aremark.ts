import { Diskresjonskoder } from '../../constants';
import { Person } from '../../models/person';

export const aremark: Person = {
    fødselsnummer: '10108000398',
    kjønn: 'M',
    geografiskTilknytning: '0118',
    alder: 42,
    navn: {
        sammensatt: 'AREMARK TESTFAMILIEN',
        fornavn: 'AREMARK',
        mellomnavn: '',
        etternavn: 'TESTFAMILIEN',
    },
    diskresjonskode: Diskresjonskoder.FORTROLIG_ADRESSE,
    statsborgerskap: 'NORSK',
    status: {
        dødsdato: undefined,
        bostatus: undefined
    },
    sivilstand: {
        value: 'GIFT',
        beskrivelse: 'Gift'
    },
    familierelasjoner: []
};