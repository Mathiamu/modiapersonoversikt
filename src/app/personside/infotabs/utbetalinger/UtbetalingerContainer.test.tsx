import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../test/Testprovider';
import UtbetalingerContainer from './UtbetalingerContainer';
import { aremark } from '../../../../mock/person/aremark';

test('Viser utbetalingercontainer med alt innhold', () => {
    const visittkortheader = renderer.create(
        <TestProvider>
            <UtbetalingerContainer fødselsnummer={aremark.fødselsnummer} />
        </TestProvider>
    );

    const json = visittkortheader.toJSON();
    expect(json).toMatchSnapshot();
});
