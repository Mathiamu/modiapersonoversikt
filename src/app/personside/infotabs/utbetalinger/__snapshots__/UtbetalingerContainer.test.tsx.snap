// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viser utbetalingercontainer med alt innhold 1`] = `
.c11 {
  border: none;
  background-color: transparent;
  border-radius: 8px;
  padding: 0;
  height: 1.2rem;
  width: 1.2rem;
}

.c11 svg {
  height: 1.2rem;
  width: 1.2rem;
}

.c11:focus {
  outline: none;
  box-shadow: 0 0 0px 3px #FFBD66;
}

.c11:hover {
  opacity: 0.8;
  cursor: pointer;
}

.c2 {
  background-color: white;
  border-radius: 4px;
  padding: 1.25rem;
  margin-bottom: .8rem;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 1.5rem 3rem 1.5rem 0;
}

.c3 > *:not(:first-child) {
  margin-top: 1.5rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 > *:first-child {
  margin-bottom: .5rem;
}

.c4 > * {
  margin-top: .5rem;
}

.c6 {
  margin-top: 1rem;
}

.c5 {
  border: none;
  margin: 0;
  padding: 0;
}

.c5 > *:first-child {
  margin-bottom: .8rem;
}

.c5 > *:last-child {
  margin-bottom: 0;
}

.c12 {
  border: none;
  padding: 0;
  height: 1.2rem;
  width: 1.2rem;
  border-radius: 0.5em;
  cursor: pointer;
  background-color: transparent;
}

.c12:hover {
  background-color: #C6C2BF;
}

.c12:focus {
  outline: none;
  box-shadow: 0 0 0px 3px #FFBD66;
}

.c9 {
  background-color: white;
  border-radius: 4px;
  padding: 1.25rem 1.25rem 2rem;
  cursor: pointer;
}

.c13 th:not(:first-child) {
  font-weight: normal;
}

.c13 th {
  text-transform: uppercase;
}

.c13 td {
  font-weight: bold;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  height: 2rem;
  padding-bottom: .2rem;
}

.c10 > *:not(:first-child) {
  margin-left: .5rem;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c18 {
  text-transform: uppercase;
}

.c17 {
  font-weight: bold;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1.25rem .6rem;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  cursor: pointer;
}

.c19 .order-first {
  -webkit-order: -1;
  -ms-flex-order: -1;
  order: -1;
}

.c19:focus {
  outline: none;
  box-shadow: 0 0 0px 3px #FFBD66;
}

.c19 > *:first-child,
.c19 > *:nth-child(2),
.c19 > *:nth-child(3) {
  height: 1.3rem;
}

.c19 > *:nth-child(3) {
  margin-bottom: .8rem;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-bottom: .2rem;
}

.c21 > *:not(:first-child) {
  margin-left: .5rem;
}

.c16 > *:first-child {
  background-color: rgba(102,203,236,0.18);
  padding: .2rem .6rem;
}

.c16 ol {
  padding: 0;
  margin: 0;
}

.c16 ol > *:not(:first-child) {
  border-top: solid 2px #efefef;
}

.c14 {
  background-color: white;
  border-radius: 4px;
  margin-top: .8rem;
}

.c14 > *:first-child {
  padding: 1.25rem;
}

.c15 {
  padding: 0 .6rem;
  margin: 0;
}

.c8 ol {
  list-style: none;
}

.c8 table {
  width: 100%;
  text-align: right;
  border-spacing: 0;
}

.c8 table * {
  padding: 0;
  margin: 0;
}

.c8 table tr > * {
  padding: 0.1rem;
  text-align: right;
}

.c8 table tr > *:first-child {
  text-align: left;
}

.c8 table tr > *:not(:first-child) {
  width: 6rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 {
  min-width: 18rem;
  -webkit-flex-basis: 18rem;
  -ms-flex-preferred-size: 18rem;
  flex-basis: 18rem;
}

.c7 {
  position: relative;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 35rem;
}

.c22 {
  text-align: right;
  position: absolute;
  top: 1.25rem;
  right: 1.25rem;
}

@media (max-width:80rem) {
  .c0 {
    display: block;
  }
}

@media not all and (max-width:80rem) {
  .c7 {
    margin-left: .8rem;
  }
}

<article
  aria-label="Utbetalinger"
  className="c0"
  role="region"
>
  <section
    className="c1"
  >
    <nav
      className="c2"
      onClick={[Function]}
    >
      <h2
        className="typo-undertittel"
      >
        Filtrering
      </h2>
      <div
        className="c3"
      >
        <form
          className="c4"
        >
          <fieldset
            className="c5"
          >
            <legend
              className="typo-etikett-liten"
            >
              Velg periode
            </legend>
            <div
              className="skjemaelement skjemaelement--horisontal"
            >
              <input
                checked={true}
                className="skjemaelement__input radioknapp"
                id="Helt tilfeldig ID"
                name="FiltreringsvalgGruppe"
                onChange={[Function]}
                type="radio"
              />
              <label
                className="skjemaelement__label"
                htmlFor="Helt tilfeldig ID"
              >
                Siste 30 dager
              </label>
            </div>
            <div
              className="skjemaelement skjemaelement--horisontal"
            >
              <input
                checked={false}
                className="skjemaelement__input radioknapp"
                id="Helt tilfeldig ID"
                name="FiltreringsvalgGruppe"
                onChange={[Function]}
                type="radio"
              />
              <label
                className="skjemaelement__label"
                htmlFor="Helt tilfeldig ID"
              >
                Inneværende år
              </label>
            </div>
            <div
              className="skjemaelement skjemaelement--horisontal"
            >
              <input
                checked={false}
                className="skjemaelement__input radioknapp"
                id="Helt tilfeldig ID"
                name="FiltreringsvalgGruppe"
                onChange={[Function]}
                type="radio"
              />
              <label
                className="skjemaelement__label"
                htmlFor="Helt tilfeldig ID"
              >
                I fjor
              </label>
            </div>
            <div
              className="skjemaelement skjemaelement--horisontal"
            >
              <input
                checked={false}
                className="skjemaelement__input radioknapp"
                id="Helt tilfeldig ID"
                name="FiltreringsvalgGruppe"
                onChange={[Function]}
                type="radio"
              />
              <label
                className="skjemaelement__label"
                htmlFor="Helt tilfeldig ID"
              >
                Egendefinert
              </label>
            </div>
          </fieldset>
          <div
            className="c6"
          >
            <button
              className="knapp knapp--standard"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Hent utbetalinger
            </button>
          </div>
        </form>
      </div>
      <div
        className="c3"
      >
        <form
          className="c4"
        >
          <p
            className="typo-etikett-liten"
          >
            Utbetaling til
          </p>
          <div
            className="skjemaelement skjemaelement--horisontal"
          >
            <input
              checked={true}
              className="skjemaelement__input checkboks"
              id="Helt tilfeldig ID"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="skjemaelement__label"
              htmlFor="Helt tilfeldig ID"
            >
              Bruker
            </label>
            <div
              aria-live="assertive"
              role="alert"
            />
          </div>
        </form>
        <form
          className="c4"
        >
          <p
            className="typo-etikett-liten"
          >
            Velg ytelse
          </p>
          <div
            className="skjemaelement skjemaelement--horisontal"
          >
            <input
              checked={true}
              className="skjemaelement__input checkboks"
              id="Helt tilfeldig ID"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="skjemaelement__label"
              htmlFor="Helt tilfeldig ID"
            >
              Sykepenger
            </label>
            <div
              aria-live="assertive"
              role="alert"
            />
          </div>
        </form>
      </div>
    </nav>
  </section>
  <section
    className="c7"
  >
    <h2
      className="typo-undertittel visually-hidden"
    >
      Filtrerte utbetalinger
    </h2>
    <div
      className="c8"
    >
      <article
        className="c9"
        onClick={[Function]}
      >
        <h2
          className="typo-undertittel"
        >
          Totalt utbetalt for perioden
        </h2>
        <nav
          className="c10"
        >
          <button
            aria-label="Skriv ut"
            className="c11"
            onClick={[Function]}
          >
            <svg
              viewBox="0 0 17 16"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink"
            >
              <path
                d="M14.8,4.9H2.2c-0.9,0-1.7,0.7-1.7,1.7v4c0,0.9,0.7,1.7,1.7,1.7h1.7v3c0,0.2,0.1,0.3,0.3,0.3h8.7	c0.2,0,0.3-0.1,0.3-0.3v-3h1.7c0.9,0,1.7-0.7,1.7-1.7v-4C16.5,5.6,15.8,4.9,14.8,4.9z M4.5,14.9h8v-4.7h-8V14.9z"
                fill="#0067C5"
                fillRule="evenodd"
              />
              <path
                d="M4.9,0.9h7.1c0.6,0,1,0.4,1,1v1.4c0,0.6-0.4,1-1,1H4.9c-0.6,0-1-0.4-1-1V1.9C3.9,1.3,4.4,0.9,4.9,0.9z"
                fill="#0067C5"
              />
            </svg>
          </button>
          <button
            aria-expanded={false}
            aria-label="Detaljer"
            className="c12"
            onClick={[Function]}
          >
            <i
              className="nav-frontend-chevron chevronboks chevron--ned"
            />
          </button>
        </nav>
        <section
          className="c13"
        >
          <span
            className="typo-undertekst"
          >
            <table>
              <thead>
                <tr>
                  <th>
                    Totalt Utbetalt
                  </th>
                  <th>
                    Brutto
                  </th>
                  <th>
                    Trekk
                  </th>
                  <th>
                    Utbetalt
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    02.12.1969 - 01.01.1970
                  </td>
                  <td>
                    13,370.00
                  </td>
                  <td>
                    -1,984.00
                  </td>
                  <td>
                    11,386.00
                  </td>
                </tr>
              </tbody>
            </table>
          </span>
        </section>
      </article>
      <article
        className="c14"
      >
        <h2
          className="typo-undertittel"
        >
          Utbetalinger
        </h2>
        <ol
          className="c15"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <li
            className="c16"
          >
            <h3
              className="typo-undertekst"
            >
              <span
                className="c17"
              >
                <span
                  className="c18"
                >
                  Mars 2017
                </span>
              </span>
            </h3>
            <ol>
              <li
                className="c19"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                tabIndex={0}
              >
                <div
                  className="c20"
                >
                  <h4
                    className="typo-undertekst-bold"
                  >
                    Sykepenger
                  </h4>
                  <p
                    className="typo-undertekst-bold"
                  >
                    11,386.00
                  </p>
                </div>
                <div
                  className="order-first c20"
                >
                  <p
                    className="typo-undertekst"
                  >
                    29. Mars 2017
                     / 
                    <span
                      className="c17"
                    >
                      Utbetalt
                    </span>
                  </p>
                  <div
                    className="c21"
                  >
                    <button
                      aria-label="Skriv ut"
                      className="c11"
                      onClick={[Function]}
                    >
                      <svg
                        viewBox="0 0 17 16"
                        xmlns="http://www.w3.org/2000/svg"
                        xmlnsXlink="http://www.w3.org/1999/xlink"
                      >
                        <path
                          d="M14.8,4.9H2.2c-0.9,0-1.7,0.7-1.7,1.7v4c0,0.9,0.7,1.7,1.7,1.7h1.7v3c0,0.2,0.1,0.3,0.3,0.3h8.7	c0.2,0,0.3-0.1,0.3-0.3v-3h1.7c0.9,0,1.7-0.7,1.7-1.7v-4C16.5,5.6,15.8,4.9,14.8,4.9z M4.5,14.9h8v-4.7h-8V14.9z"
                          fill="#0067C5"
                          fillRule="evenodd"
                        />
                        <path
                          d="M4.9,0.9h7.1c0.6,0,1,0.4,1,1v1.4c0,0.6-0.4,1-1,1H4.9c-0.6,0-1-0.4-1-1V1.9C3.9,1.3,4.4,0.9,4.9,0.9z"
                          fill="#0067C5"
                        />
                      </svg>
                    </button>
                    <button
                      aria-expanded={false}
                      aria-label="Detaljer"
                      className="c12"
                      onClick={[Function]}
                    >
                      <i
                        className="nav-frontend-chevron chevronboks chevron--ned"
                      />
                    </button>
                  </div>
                </div>
                <div
                  className="c20"
                >
                  <p
                    className="typo-undertekst"
                  >
                    09.11.2016 - 31.10.2017
                  </p>
                  <p
                    className="typo-undertekst"
                  >
                    
                  </p>
                </div>
                <p
                  className="typo-undertekst"
                >
                  Utbetaling til: 
                  AREMARK TESTFAMILIEN
                </p>
              </li>
            </ol>
          </li>
        </ol>
      </article>
    </div>
    <div
      className="c22"
    >
      <p
        className="typo-undertekst"
      >
        <a
          className="lenke"
        >
          Meldinger/utbetalinger i Arena
        </a>
      </p>
    </div>
  </section>
</article>
`;
